### ISSUE TITLE
Is this DTH working right now? 

### ISSUE THREAD

**Issue by kianrafiee** [jlboygenius/SmartThingsSertaBedControl#1]
Was wondering if this can actually be used with ST/Alexa in current state? Is there a way to find the correct Alexa commands? I read the code in your git and it doesn't show any command words. 

This is awesome that your developing this.

**Comment by jlboygenius** [comment]
Yes, the code here works. I meant to put together a blog post about it but never finished it.
https://hackaday.io/project/41172-alexasmartthings-controlled-adjustable-bed

I added in the missing code definition files which have the command codes necessary to talk to the bed.
I also added in the SmartThings Device Handler code. Particle device ID and auth key are hard coded. Could be updated so that each device has it's own unique key, if you need to control multiple beds.

**Comment by kianrafiee** [comment]
Kudos for figuring this stuff out! People like you are why we get improvements in quality of life all the time.

Some questions:

1) Do you have to have an arduino/other device to communicate with the base and the smartthings Hub? Or is the smartthings Hub with the DTH enough?

2) Doesn't the Serta base you currently use have a bluetooth adapter hooked up to the emergency power off module of the base and thats how it communicates the raise/lower commands? In that case, could you not create a DTH that searches for bluetooth signals from the module? Or is that sort of what your doing right now?

3) So right now its acting like a switch in smartthings. Does that mean you can only turn massage on and off? Or can you assign different behaviors to the on/off switches such as On being Zero G and Off being Flat base. 

4) What would it take for you to add more words/commands to the Alexa skill such as "Alexa, raise/lower head on base" or "Alexa, go to Zero-G on base" or "Alexa, start a full body massage on base"? I'm guessing you might need to create a real Alexa skill for that stuff? I had some prior experience with skill coding but it was very basic stuff using there templates. I wish I could help but wouldnt know where to start right now unless you already have some ideas.

5) So looking at the Serta iComfort Motion Perfect manual, you can see there is a bluetooth module that ataches to the emergecy power off via a DIN connection which I was explaining earlier. I have a Beautyrest Smart Motion 3.0 which is the same base because yours and mine are both actually manufactured by Ergomotion. So we both have the emergency power off module with DIN connnection. How likely do you think it would be that I could just buy the bluetooth module from Ergomotion and it would work? This assumes you are using bluetooth it communicate with your base.

6) Have you tried using the Serta app to control your base form a phone? That may point you to how it communicates over bluetooth with the module on the base.

**Comment by jlboygenius** [comment]
1: No other things involved. Particle communicates over SPI to CC2500 transciever, which sends commands to the Serta adjustable base (imitating the stock remote). SmartThings communicates to the Particle. Alexa communicates to SmarthThings.

2: My Serta base is from ~2012. There was no bluetooth module back then.  There are a few different versions of the serta base controller. I'm not sure how other versions may work. I use a CC2500 transceiver module, which also uses 2.4GHZ frequency, but in a different way than bluetooth. 

3: In SmartThings, it can control all the buttons of the bed control, except massage. (Those buttons could easily be added.) SmartThings exposes it as a switch to Alexa, which means it only supports 2 commands: ON and OFF. I setup ON to trigger the Louge button, and OFF to trigger the Flat button. You could change the code to do something different.

4: I think I read that Particle had a new integration with alexa to create more customized phrases. The Alxea->Particle Integration came out after I finished this project. My project uses SmartThings to do the Alexa integration. I built this so that SmartThings could control it. Alexa integration was a bonus and easy after that.  I think more complex phrases and settings would require use of the Alexa->Particle skill. I haven't looked into it.

5: The remote for your unit looks very similar to mine. I imagine that my system would work with yours. I do not have the bluetooth module, and this project does not use bluetooth.  This project uses a CC2500, which can be found on ebay for <$5.  Search ebay for CC2500. They are cheap, and very small. 
Depending on the CC2500 module, soldering is required to connect the wires from the CC2500 to the Particle. There are 6 wires.  


6: No bluetooth. My bed base is ~6 years old. Bluetooth wasn't an option back then.  I added notes to the readme with a picture of the remote that my bed uses.
